h/static/scripts/installer-controller.js

```diff
@@ -0,0 +1,34 @@
+// Utility functions used on the page that presents the Hypothesis
+// extension and bookmarklet installers
+
+var uaDetect = require('./ua-detect');
+
+function showSupportedInstallers(rootElement) {
+  function showIf(selector, cond) {
+    var elements = rootElement.querySelectorAll(selector);
+    for (var i = 0; i < elements.length; i++) {
+      if (cond) {
+        elements[i].classList.remove('is-hidden');
+      } else {
+        elements[i].classList.add('is-hidden');
+      }
+    }
+  }
+
+  // check for Chrome extension support. We also check here for !mobile
+  // so that the page is correct when testing in Chrome dev tools with
+  // mobile device simulation enabled
+  var offerChromeInstall = uaDetect.chromeExtensionsSupported &&
+                           !uaDetect.isMobile;
+  var offerBookmarkletInstall = !offerChromeInstall &&
+                                !uaDetect.isMobile &&
+                                !uaDetect.isMicrosoftEdge;
+
+  showIf('.js-install-chrome', offerChromeInstall);
```

Comment:
It make make more sense to take a Modernizr-like approach here and add "feature" classes to `document.body` which can then be used to show/hide parts of the page from CSS, as opposed to having to explicitly run `showIf` for each element.

If you need to hide things until you can detect features, but want them shown by default if JS fails to load, you can use the old chestnut of putting `document.documentElement.classList.add('js')` before any other scripts in the `<head>` to avoid a FOUC.